### DISCUSSION TITLE
Detect if a PDF has a password

### DISCUSSION THREAD

**Discussion by hamelpierreluc** [empira/PDFsharp#189] [Q&A]
Hello, I have some PDF with a password, so I can't open it with Adobe et see the PDF. But in .net code with PDFsharp, I can open the PDF and I don't see a properties to detect the password. This is a problem for us because we would detect the password and reject the PDF.

Do you know this kind of PDF? Do you have a hint for me ?

**Comment by packdat** [comment]
Have not tested it myself but a first naive approach that comes into mind is using an overload of `PdfReader.Open` that takes a `PdfPasswordProvider` as an argument.
When the password-provider is invoked, it means the library has detected that a password is needed to open the document and it uses the password-provider to ask _you_ for the password.
You don't have to supply an actual password but you can use the fact _that_ the provider was invoked to tell the PDF is password-protected.
Something like this:
```
var isPasswordProtected = false;
var document = PdfReader.Open(filePath, PdfDocumentOpenMode.Modify, passwordArgs =>
{
    passwordArgs.Abort = true;  // do not open the document
    isPasswordProtected = true;
});
if (isPasswordProtected)
{
    // discard the documemt
}
```

**Reply by hamelpierreluc** [reply]
The issue is that the PDF file opens without any problem using PdfReader.Open, but if I try to open it with Acrobat Reader or any other tool, a prompt appears asking for a password.

**Reply by TH-Soft** [reply]
Have you tried it with the *PdfDocumentOpenMode.Modify* flag?

**Reply by hamelpierreluc** [reply]
Sure : 

```
var isPasswordProtected = false;
using var pdf = PdfReader.Open(fileStream, PdfDocumentOpenMode.Modify, passwordArgs =>
{
    passwordArgs.Abort = true;
    isPasswordProtected = true;
});
if (pdf.PageCount == 0)
{
    return false;
}
```
`

You have to know that it's not with all pdf with password, the majority of our pdf with password, pdfReader.Open goes in exception like we expect.
